backend/src/test/java/ee/tlu/forum/controller/PostControllerTest.java

```diff
@@ -0,0 +1,244 @@
+        User user = new User(1L,
+                "user1",
+                "test1@test.com",
+                "aaa",
+                new ArrayList<>(),
+                "",
+                "",
+                new ArrayList<>(),
+                new ArrayList<>());
+        Thread t1 = new Thread(1L, "content", "title", user, new ArrayList<>());
+        Post p1 = new Post(1L, "text", user, t1);
+
+        Map<String, String> form = new HashMap<>();
+        form.put("title", t1.getTitle());
+        String formJson = new ObjectMapper().writeValueAsString(form);
+
+        when (postService.editPost(any()))
+                .thenReturn(p1);
+
+        RequestBuilder requestBuilder = MockMvcRequestBuilders.patch("/api/post/edit")
+                .accept(MediaType.APPLICATION_JSON)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(formJson);
+
+        // when then
+        mockMvc.perform(requestBuilder)
+                .andExpect(MockMvcResultMatchers.status().is(200))
+                .andExpect(MockMvcResultMatchers.content().contentType(MediaType.APPLICATION_JSON))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.id", Matchers.is(1)))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.text", Matchers.is("text")));
+    }
+
+    @Test
+    @DisplayName("Returns a list of posts belonging to a user by user id - GET /api/posts/userid/{id}")
+    void getPostsByUserId() throws Exception {
+        // given
+        User user = new User(1L,
+                "user1",
+                "test1@test.com",
+                "aaa",
+                new ArrayList<>(),
+                "",
+                "",
+                new ArrayList<>(),
+                new ArrayList<>());
+        Thread t1 = new Thread(1L, "content", "title", user, new ArrayList<>());
+        Post p1 = new Post(1L, "text", user, t1);
+        Post p2 = new Post(2L, "text", user, t1);
+        Collection<Post> posts = user.getPosts();
+        posts.add(p1);
+        posts.add(p2);
+        user.setPosts(posts);
+
+        when (postService.getAllPostsByUserId(any()))
+                .thenReturn((List<Post>) user.getPosts());
+        // when then
+        mockMvc.perform(MockMvcRequestBuilders.get("/api/posts/userid/1"))
+                .andDo(MockMvcResultHandlers.print())
+                .andExpect(MockMvcResultMatchers.status().is(200))
+                .andExpect(MockMvcResultMatchers.content().contentType(MediaType.APPLICATION_JSON))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.size()", Matchers.is(2)))
+                .andExpect(MockMvcResultMatchers.jsonPath("$[0].id", Matchers.is(1)))
+                .andExpect(MockMvcResultMatchers.jsonPath("$[1].id", Matchers.is(2)))
+                .andExpect(MockMvcResultMatchers.jsonPath("$[0].text", Matchers.is("text")));
+    }
+
+    @Test
+    @DisplayName("Returns a list of posts belonging to a user by username - GET /api/posts/userid/{id}")
```

Comment:
Shouldn't the parameter at the end here be username or something?

Reply:
That's true - the fallacy of copy/paste :P

Reply:
I'll approve but perhaps change this little bit when you next get to the Posts controller 😄